### PR TITLE
Added TurningOffLightsEventArgs, added overload for Player::UseItem and added overload for Player::Get(Collider)

### PR DESCRIPTION
**Pull request by Monaldcry7788**

### CODE REVIEW

**Comment by louis1706** [comment]
Can you add the TurningOffLightsEventArgs::Room ?

**Comment by Monaldcry7788** [comment]
no

**Comment by Monaldcry7788** [comment]
It's useless, because the target isn't the room, but the controller

**Comment by Monaldcry7788** [comment]
With the controller, you can get the rooms

**Comment by louis1706** [comment]
> With the controller, you can get the rooms

Oh okay

File: Exiled.Events/EventArgs/TurningOffLightsEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        /// <param name="isAllowed"><inheritdoc cref="IsAllowed"/></param>
+        public TurningOffLightsEventArgs(FlickerableLightController flickerableLightControllerHandler, float duration, bool isAllowed = true)
+        {
+            FlickerableLightControllerHandler = flickerableLightControllerHandler;
+            Duration = duration;
+            IsAllowed = isAllowed;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="FlickerableLightControllerHandler"/>.
```
Comment (Ficus-x) [inline Exiled.Events/EventArgs/TurningOffLightsEventArgs.cs:29]: I think you forgot `Instance` after `Gets the see cref="FlickerableLightComtrollerHandler"/>`

File: Exiled.Permissions/Exiled.Permissions.csproj [hunk 2]
```diff
@@ -27,6 +27,7 @@
     <Reference Include="Mirror" HintPath="$(EXILED_REFERENCES)\Mirror.dll" Private="false" />
     <Reference Include="UnityEngine.CoreModule" HintPath="$(EXILED_REFERENCES)\UnityEngine.CoreModule.dll" Private="false" />
     <Reference Include="NorthwoodLib" HintPath="$(EXILED_REFERENCES)\NorthwoodLib.dll" Private="false" />
+    <Reference Include="UnityEngine.PhysicsModule" HintPath="$(EXILED_REFERENCES)\UnityEngine.PhysicsModule.dll" Private="false" />
```
Comment (IRacle1) [inline Exiled.Permissions/Exiled.Permissions.csproj:30]: why

Comment (Monaldcry7788) [reply Exiled.Permissions/Exiled.Permissions.csproj:30]: For the Player.Get(Collider) i thinks. Compiler give me a problem in build. Exiled.permission needs Physics, so i added it

Comment (IRacle1) [reply Exiled.Permissions/Exiled.Permissions.csproj:30]: fair

**Review by rus-skill-issue — APPROVED** [review]

File: Exiled.Events/Patches/Events/Map/TurningOffLights.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+            newInstructions.InsertRange(0, new CodeInstruction[]
+            {
+                new(OpCodes.Ldarg_0),
+                new(OpCodes.Ldarg_1),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Newobj, GetDeclaredConstructors(typeof(TurningOffLightsEventArgs))[0]),
+                new(OpCodes.Dup),
+                new(OpCodes.Dup),
+                new(OpCodes.Stloc_S, ev.LocalIndex),
```
Comment (Jesus-QC) [inline Exiled.Events/Patches/Events/Map/TurningOffLights.cs:42]: Instead of dupping use this local

**Review by Jesus-QC — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Map/TurningOffLights.cs (hunk 3 above)
Comment (IRacle1) [reply Exiled.Events/Patches/Events/Map/TurningOffLights.cs:42]: why?

Comment (IRacle1) [reply Exiled.Events/Patches/Events/Map/TurningOffLights.cs:42]: why?

Comment (Jesus-QC) [reply Exiled.Events/Patches/Events/Map/TurningOffLights.cs:42]: I mean, is actually the same

**Review by Jesus-QC — APPROVED** [review]

**Review by rus-skill-issue — APPROVED** [review]

**Review by Jesus-QC — APPROVED** [review]

**Pull request merged**